packages/plugin-user/src/user.js

```diff
@@ -173,6 +183,51 @@ const User = SparkPlugin.extend({
   },
 
   /**
+   * Sets password status on token
+   * @param {Boolean} value
+   * @returns {undefined}
+   */
+  setPasswordStatus(value) {
+    this.spark.credentials.setPasswordStatus(value);
+  },
+
+  /**
+   * Updates a user's password with spark.
+   * @param {Object} options
+   * @param {string} options.password (required)
+   * @param {string} options.userId (required)
+   * @returns {Promise} Resolves with complete user object containing new password
+   */
+  setPassword(options) {
+    options = options || {};
+    if (!options.password) {
+      return Promise.reject(new Error(`\`options.password\` is required`));
+    }
+    if (!options.userId) {
```

Comment:
should get called after device registration. 👍 changed it to `this.spark.device.userId`